### DISCUSSION TITLE
VScode Quarto render pathway error: index.qmd not found

### DISCUSSION THREAD

**Discussion by xinhuaChenW** [quarto-dev/quarto-cli#2567] [Q&A] [answered]
Please help! 
When rendering, how to let Quarto Preview type in a pathway of .qmd file with a 'forward slash' instead of 'backforward slash' ? or, in another way, how to make Quarto to choose PowerShell as default terminal to run a line of code with 'backforward slash' ( I assume PowerShell can run the code as it identifys 'backforward slash'.)?

The following is how the problem happened.
When rendering in VScode (by clicking on the 'render button'), Quarto Preview typed in `quarto preview d:\githubRepos\quarto\index.qmd --to html --no-browser --no-watch-inputs` in Git-Bash terminal and throw an error like `ERROR: d:githubReposquartoindex.qmd not found`.

What I have tried.
(1) I switched default terminal as PowerShell, but it still didn't work. The terminal switch back to Git-Bash itself and throw the same error `ERROR: d:githubReposquartoindex.qmd not found`.
(2) I replace the '\' with '/' and typed in `quarto preview d:/githubRepos/quarto/index.qmd --to html --no-browser --no-watch-inputs`
in Git-Bash terminal, Quarto render works well.
So, the problem must be the forward- and backforward- slash thing. 

The error looks like:
<img width="512" alt="image" src="https://user-images.githubusercontent.com/63830206/191872964-192ec54d-c498-43df-997e-ef475662ece9.png">

Working environment:
Windows 10 LST 1809
VScode 1.71.2
Quarto CLI 1.1.251(Windows)

**Comment by mcanouil** [comment]
Quarto VSCode is using your default terminal.
When switching your default terminal to PowerShell (maybe a restart is needed), the quarto-vscode extension should use PowerShell.
I'll try later on a Windows laptop to see if I can reproduce and help more concretely.

**Reply by xinhuaChenW** [reply]
Thanks for your reply! According to your reply, I tried switch the default terminal to PowerShell and then restarted VSCode. Indeed, just like what you'v said, the default terminal changed to PowerShell, as a PowerShell terminal is established when click on the '+' sign. However, when click on the 'Render' button, the terminal still is swicthed back to Git-Bash to run the code.

I ran into a solution and luckily it worked.
The solution will be in a new answer.

Again, Thank you !!

**Comment by xinhuaChenW** [comment, marked as answer]
The problem has something to do with the VSCode default terminal setting. 
(This solution hit me by chance and happily it works.)

What I tried later.

 (1) File → Preferences → Settings menu → click the <img width="31" alt="image" src="https://user-images.githubusercontent.com/63830206/192081096-9803054e-be69-4923-b297-532971ca3e3c.png"> to open the 'setting.json' file.

(2) change the `"terminal.integrated.defaultProfile.windows":"Git Bash", `  to  `"terminal.integrated.defaultProfile.windows": "Windows PowerShell",`
<img width="516" alt="image" src="https://user-images.githubusercontent.com/63830206/192081204-9c3f68e3-023b-4fad-81cf-7d48eebaf636.png">

and then IMPORTANTLY!

(3) comment out the following line of code:

`"terminal.integrated.shell.windows": "C:\\Program Files\\Git\\bin\\bash.exe",`
<img width="606" alt="image" src="https://user-images.githubusercontent.com/63830206/192081276-b020c1c8-fcb8-464b-ad34-d73c2f272571.png">

Why? 

Because, step (2) will not come into effect unless `"terminal.integrated.shell.windows": "C:\\Program Files\\Git\\bin\\bash.exe",` is commented out. 
VSCode CURRENTLY gives higher priority to `"terminal.integrated.shell.windows": "C:\\Program Files\\Git\\bin\\bash.exe",` than  `"terminal.integrated.defaultProfile.windows": "Windows PowerShell",`. 

Setting default terminal by the code  `"terminal.integrated.shell.windows": ` is deprecated and VSCode recommends `"terminal.integrated.defaultProfile.windows": `. However, CURRENTLY, VSCode will still the former higher priority if it exits.
